test/e2e/utils.go

```diff
@@ -0,0 +1,252 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
+	oputil "github.com/projectcontour/contour-operator/util"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = operatorv1alpha1.AddToScheme(scheme)
+}
+
+func newClient() (client.Client, error) {
+	opts := client.Options{
+		Scheme: scheme,
+	}
+	kubeClient, err := client.New(config.GetConfigOrDie(), opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kube client: %v\n", err)
+	}
+	return kubeClient, nil
+}
+
+func newContour(t *testing.T, ctx context.Context, cl client.Client, name, ns string) {
```

Comment:
Everything in this file that takes a `testing.T` should first call `t.Helper()`. This makes sure that the test failure is attributed to the caller, not the the helper function, which makes it much easier to triage test failures.